### DISCUSSION TITLE
Domain Events - get notified outside of application layer

### DISCUSSION THREAD

**Discussion by Eagleleader** [jasontaylordev/CleanArchitecture#438] [Q&A] [answered]
Hi all,

I've tried to get notified of the domain event in an API using the Application layer.
I created a SignalR hub and an Handler like this inside the API:

```
public class MessagesHub : Hub
	{
		public static string HubUrl => Enums.HubUrl;
	}

	public class SentOrderCreatedEventHubHandler : INotificationHandler<DomainEventNotification<SentOrderCreatedEvent>>
	{
		private readonly IHubContext<MessagesHub> _hubContext;
		private readonly ILogger<SentOrderCreatedEventHubHandler> _logger;

		public SentOrderCreatedEventHubHandler(IHubContext<MessagesHub> hubContext, ILogger<SentOrderCreatedEventHubHandler> logger)
		{
			_hubContext = hubContext;
			_logger = logger;
		}

		public async Task Handle(DomainEventNotification<SentOrderCreatedEvent> notification, CancellationToken cancellationToken)
		{
			_logger.LogInformation($"Sending message from [{nameof(SentOrderCreatedEventHubHandler)}] to clients: {EnumsHelper.GetValueFromDescription(Events.NewSentOrderCreated)}: {notification.DomainEvent.SentOrder}");
			await _hubContext.Clients.All.SendAsync(EnumsHelper.GetValueFromDescription(Events.NewSentOrderCreated), notification.DomainEvent, cancellationToken);
		}
	}
```

I'm new to this approach and I don't see what I'm doing wrong here.
Would you be kind enough to help me?

**Comment by Eagleleader** [comment, marked as answer]
Answer is : just add the MediatR into the services of the API :
`services.AddMediatR(Assembly.GetExecutingAssembly());`
As my API currently uses the application layer and uses the Dependency Injection configured into that APP Layer, everything is communicating all together.

**Reply by artworkad** [reply]
In your example you are injecting the hub into SentOrderCreatedEventHubHandler which is in the application layer. You application layer now depends on signlar. Would'nt it be more appropriate to use an interface in the application layer and create an implementation in the infrastructure layer using signlar?
